src/ldesify.ts

```diff
@@ -189,3 +196,90 @@ export function ldesify(
         return writer.push(st);
     });
 }
+
+export function ldesify_sds(
+    reader: Stream<string>,
+    writer: Writer<string>,
+    statePath: string | undefined,
+    sourceStream: Term | undefined,
+    targetStream: Term,
+    modifiedPathM?: Term, // This are the things that are added to the new entities, not necessarily related to the real objects
+    isVersionOfPathM?: Term,
+) {
+    const modifiedPath: Term = modifiedPathM || MODIFIED;
+    const versionPath: Term = isVersionOfPathM || IS_VERSION_OF;
+
+    let cache: { [key: string]: string } = {};
+    if (statePath) cache = getPrevState(statePath);
+
+    Cleanup(() => {
+        if (statePath) {
+            writeFileSync(statePath, JSON.stringify(cache), {
+                encoding: "utf8",
+            });
+        }
+    });
+    reader.on("end", () => {
+        if (statePath) {
+            writeFileSync(statePath, JSON.stringify(cache), {
+                encoding: "utf8",
+            });
+        }
+    });
+
+    const extractor = new Extractor(new CBDShapeExtractor(), sourceStream);
+    reader.on("data", async (x) => {
+        const quads = new Parser().parse(x);
+
+        const records = await extractor.parse_records(quads);
+
+        for (const rec of records) {
+            if (rec.data.quads.length === 0) continue;
+
+            const hash = await canonize(rec.data.quads, {
+                algorithm: "RDFC-1.0",
+            });
+            if (cache[rec.data.id.value] === hash) continue;
+
+            cache[rec.data.id.value] = hash;
+
+            const date = new Date();
+            const id = new NamedNode(rec.data.id.value + date.getTime());
```

Comment:
Perhaps you also want to add a `#` in between the value and the time, as happens in the basic ldesify as well https://github.com/rdf-connect/sds-processors-ts/blob/933a8cb9a5dce73b682a82c646b289700badd5d2/src/ldesify.ts#L140

Otherwise, I guess that values like `https://example.org` might result in unwanted results like `https://example.org123456789` instead of a dereferenceable `https:/example.org#123456789`